src/credit-card-number-field.tsx

```diff
@@ -17,39 +17,57 @@ import {createUseStyles} from './jss';
 
 import type {CommonFormFieldProps} from './text-field-base';
 import type {CardOptions} from './utils/credit-card';
+import {useRifm} from 'rifm';
+import {createChangeEvent} from './utils/dom';
+import {combineRefs} from './utils/common';
 
-const CreditCardInput = ({inputRef, defaultValue, value, maxLength, onInput, ...rest}: any) => {
-    // Naive implementation, some issues in cursor position when editing
-    const format = (s?: string) => {
-        const chars = String(s ?? '')
-            .replace(/[^\d]/g, '')
-            .slice(0, 16)
-            .split('');
+const format = (s?: string) => {
+    const sanitizedNumber = String(s ?? '').replace(/[^\d]/g, '');
+    return sanitizedNumber.match(/.{1,4}/g)?.join(' ') ?? sanitizedNumber;
+};
```

Comment:
I've changed the code of format function, now it is based on `formatIban` implementation, to keep them similar: https://github.com/Telefonica/mistica-web/blob/master/src/iban-field.tsx#L76-L79